ngraph/core/src/op/roll.cpp

```diff
@@ -0,0 +1,107 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <ngraph/validation_util.hpp>
+
+#include "itt.hpp"
+#include "ngraph/op/roll.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v7::Roll, "Roll", 7);
+
+op::v7::Roll::Roll(const Output<Node>& data, const Output<Node>& shift, const Output<Node>& axes)
+    : Op({data, shift, axes})
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v7::Roll::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v7_Roll_validate_and_infer_types);
+
+    const auto& shift_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          shift_et.is_dynamic() || shift_et.is_integral_number(),
+                          "Shift must have an integral number element type.");
+
+    const auto& axes_et = get_input_element_type(2);
+    NODE_VALIDATION_CHECK(this,
+                          axes_et.is_dynamic() || axes_et.is_integral_number(),
+                          "Axes must have an integral number element type.");
+
+    const auto& data_pshape = get_input_partial_shape(0);
+    const auto& shift_pshape = get_input_partial_shape(1);
+    const auto& axes_pshape = get_input_partial_shape(2);
+    const auto& data_rank = data_pshape.rank().get_length();
+    const auto& shift_rank = shift_pshape.rank().get_length();
+    const auto& axes_rank = axes_pshape.rank().get_length();
```

Comment:
Rank can be dynamic. So the `get_length()` will fail. Please, add a check that the rank is not dynamic before getting its length

Reply:
Please, add unit tests for this case.